recipe/conda_build_config.yaml

```diff
@@ -51,18 +48,13 @@ m2w64_cxx_compiler:            # [win]
 m2w64_fortran_compiler:        # [win]
   - m2w64-toolchain            # [win]
 
-cuda_compiler:
-  - None
-  - nvcc                       # [(linux or win64) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
-  - cuda-nvcc                  # [(linux or win64) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
```

Comment:
do we really need to use this key? could we have users simply depend on cuda-nvcc? I likely don't understand the downstream implications.

Reply:
Users shouldn't depend on `cuda-nvcc` but continue to depend on `{{ compiler("cuda") }}`. For that, we need to set `cuda_compiler` in the pinning, also in a 12+ world